odoo-openmrs/src/main/java/com/ozonehis/eip/odoo/openmrs/handlers/PartnerHandler.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2024, Ozone HIS <info@ozone-his.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.ozonehis.eip.odoo.openmrs.handlers;
+
+import static java.util.Arrays.asList;
+
+import com.ozonehis.eip.odoo.openmrs.Constants;
+import com.ozonehis.eip.odoo.openmrs.client.OdooClient;
+import com.ozonehis.eip.odoo.openmrs.client.OdooUtils;
+import com.ozonehis.eip.odoo.openmrs.mapper.odoo.PartnerMapper;
+import com.ozonehis.eip.odoo.openmrs.model.Partner;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.ProducerTemplate;
+import org.hl7.fhir.r4.model.Patient;
+import org.openmrs.eip.EIPException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Setter
+@Component
+public class PartnerHandler {
+
+    @Autowired
+    private OdooClient odooClient;
+
+    @Autowired
+    private PartnerMapper partnerMapper;
+
+    public Partner partnerExists(String partnerRefID) {
+        Object[] records = odooClient.searchAndRead(
+                Constants.PARTNER_MODEL, List.of(asList("ref", "=", partnerRefID)), Constants.partnerDefaultAttributes);
+        if (records == null) {
+            throw new EIPException(
+                    String.format("Got null response while searching for Partner with reference id %s", partnerRefID));
+        } else if (records.length == 1) {
+            log.debug("Partner exists with reference id {} record {}", partnerRefID, records[0]);
+            return OdooUtils.convertToObject((Map<String, Object>) records[0], Partner.class);
+        } else if (records.length == 0) {
+            log.warn("No Partner found with reference id {}", partnerRefID);
+            return null;
+        } else {
+            log.warn("Multiple Partners exists with reference id {}", partnerRefID);
+            throw new EIPException(String.format("Multiple Partners exists with reference id%s", partnerRefID));
+        }
+    }
+
+    public int ensurePartnerExistsAndUpdate(ProducerTemplate producerTemplate, Patient patient) {
```

Comment:
May be you can rename this to createorUpdate()

Reply:
Renamed to `createOrUpdatePartner`